internal/extsvc/gerrit/client.go

```diff
@@ -45,24 +56,31 @@ func NewClient(urn string, url *url.URL, creds *AccountCredentials, httpClient h
 		Password: creds.Password,
 	}
 
-	return &Client{
+	return &client{
 		httpClient: httpClient,
 		URL:        url,
 		rateLimit:  ratelimit.DefaultRegistry.Get(urn),
 		auther:     auther,
 	}, nil
 }
 
-func (c *Client) WithAuthenticator(a auth.Authenticator) *Client {
-	return &Client{
+func (c *client) WithAuthenticator(a auth.Authenticator) (Client, error) {
+	switch a.(type) {
+	case *auth.BasicAuth, *auth.BasicAuthWithSSH:
+		break
+	default:
+		return nil, errors.Errorf("authenticator type unsupported for Azure DevOps clients: %s", a)
+	}
```

Comment:
Nit: this might be just me, but since this is basically returning an error for unsupported cases, I would not use a switch, but a plain if condition instead. Seems much shorter to me and more readable.

```suggestion
	if a.(type) != *auth.BasicAuth && a.(type) != *auth.BasicAuthWithSSH {
		return nil, errors.Errorf("authenticator type unsupported for Azure DevOps clients: %s", a)
	}
```

Reply:
Unfortunately, a.(type) only works within switch statements, so if you wanted to avoid the switch block you would need to check each type in a separate if statement, and I think the switch is a bit cleaner.

Reply:
TIL. thank you for the explanation 👍